test/unit/three_scale/redis_config_test.rb

```diff
@@ -78,5 +74,18 @@ class RedisConfigTest < ActiveSupport::TestCase
       ]
       assert_equal expected_sentinels, config.sentinels
     end
+
+    test 'sets redis client id to nil when id not provided' do
+      config = RedisConfig.new(url: 'redis://my-redis/1')
+
+      assert_nil config.id
+      assert config.key? :id
+    end
+
+    test 'sets redis client id to nil when id is set explicitly' do
```

Comment:
I approved but I'm a little bit conflicted now. If our tests are kind of a documentation. What will this test tell me afte 1 month when I don't remember this PR? That it should always be nil. But maybe at that time some customer requested to set it in config. So I will have no idea why we decided that it should be `nil` even if it was set.

So I was happier with the previous version where such behavior was undefined. For lets say not being decided yet. Now this test is testing current behavior but it doesn't give understanding of the situation or improvement in any way I can see.

So it is fine with me to merge but IMO better without it. Or with more explanation why do we test this.

Reply:
I don't have a strong opinion on this point, I am OK either way. So, I'll let you and @josemigallas come to an agreement around this :wink:

Reply:
Without out, after 1 month when we forget this PR, someone might be tempted to "fix" this forced nil value. This test is nothing but a fail safe. I suggest to add a comment on top of it linking to the JIRA ticket. Example:
```
# Fixes a compatibility issue with sidekiq in GCP. See https://issues.redhat.com/browse/MGDAPI-5164
test 'redis client id cannot be set' do
```

Reply:
@josemigallas , there already was a test to ensure without setting this explicitly it should be null. IMO if one sets it somehow, then it is fine to be something else on their responsibility that their server does support this.

Reply:
And maybe that note should be above the `client.id=nil` code line, not in the test.

Otherwise we are saying that client id will break everything which is not true. It only breaks in environments where Redis was broken by the vendor.

Reply:
I agree with @josemigallas' comment and that it should be in `app/lib/three_scale/redis_config.rb`, (not above the test) for easier access.

Reply:
> @josemigallas , there already was a test to ensure without setting this explicitly it should be null. IMO if one sets it somehow, then it is fine to be something else on their responsibility that their server does support this.

> And maybe that note should be above the `client.id=nil` code line, not in the test.
> 
> Otherwise we are saying that client id will break everything which is not true. It only breaks in environments where Redis was broken by the vendor.

OK you have a point there, but then again... not settings a client id is our responsibility? We are tweaking redis when it is sidekiq's fault, maybe we should fix _that_. _By default, Sidekiq sets a client ID for Redis_. And also, it depends on environment where _Redis was broken by the vendor_ so maybe we should target those environments?

Reply:
@josemigallas I don't understand your reply very well. Let me try to explain more thoroughly.

My understanding is that normal Redis including one provided by Amazon, supports this command, so administrator can see connected client ids and potentially see which one creates troubles and fix it. Some Redis servers (including  one provided by Google) does not support that.

So my assumption is that some day a customer may ask us - hey, allow us to set client ID to monitor what crap you're doing.

And then looking at the test, one may not understand that we disabled client ids only for the purposes of particular providers. But may think that client ID breaks who knows what, maybe whole sidekiq.

So my concern is to make it crystal clear that this change was done to fix some providers and not something we really want to enforce in all circumstances. But just it made sense to disable it in all cases because we don't see it likely for anybody to need this.

Reply:
I mean this problem only affects RHOAM on Google cloud but we are hardcoding it into our redis_config, affecting everyone.

If someone want's to deploy RHOAM on Google cloud or any other platform with an incompatible redis client, then _only they_ should specifically disable client id.

> some day a customer may ask us - hey, allow us to set client ID to monitor what crap you're doing.

In that case, they should simply be able to make a change in their config, no? Set a `client_name=habba babba`.

Reply:
I agree with what you're saying @josemigallas 

We discussed the option of making the client configurable [here](https://github.com/3scale/porta/pull/3197/commits/5b431baa8762bbb5374773f9c5b46fc1bf7a8adf#r1097740972) (the initial approach was to just disable client ID through config https://github.com/3scale/porta/pull/3197/commits/5b431baa8762bbb5374773f9c5b46fc1bf7a8adf)

But that would be a new feature, we didn't have it before, it was just Sidekiq default name and that's it.
Making client ID configurable would require some extra work, including work on the operator, and nobody has ever asked us about this feature. I doubt anyone ever will, but if they do - we'll certainly add it, not a big deal. But why doing extra effort for something people might not even be interested in?